DEVELOPER_README.md

```diff
@@ -4,7 +4,38 @@ Installation of WordPress and most plugins is handled with Composer. If
 you aren't familiar with Composer, you can get instructions on installation
 and how to use it here (<https://getcomposer.org/>).
 
-### Getting Started
+## Getting Started (Webonary cloud api integration development only)
+
+If you are working on Webonary integration with the Cloud API backend only, you can use
+[wp-env](https://www.npmjs.com/package/@wordpress/env)
+for local development using [Docker](https://www.docker.com/).
+
+To start, clone this full repo and install [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) packages and install Wordpress plugins through [composer](https://getcomposer.org/).
+
+```bash
+git clone https://github.com/sillsdev/webonary.git
+
+npm install
+
+cd webonary
+composer install
```

Comment:
Do I need to install some extensions?

I installed php with `sudo apt-get install php`
then installed composer with https://getcomposer.org/doc/00-intro.md#globally

```
ubuntu@ubuntu:~/webonary$ composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - phpoffice/phpspreadsheet is locked to version 1.23.0 and an update of this package was not requested.
    - phpoffice/phpspreadsheet 1.23.0 requires ext-dom * -> it is missing from your system. Install or enable PHP's dom extension.

To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.4/cli/php.ini
    - /etc/php/7.4/cli/conf.d/10-opcache.ini
    - /etc/php/7.4/cli/conf.d/10-pdo.ini
    - /etc/php/7.4/cli/conf.d/20-calendar.ini
    - /etc/php/7.4/cli/conf.d/20-ctype.ini
    - /etc/php/7.4/cli/conf.d/20-exif.ini
    - /etc/php/7.4/cli/conf.d/20-ffi.ini
    - /etc/php/7.4/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.4/cli/conf.d/20-ftp.ini
    - /etc/php/7.4/cli/conf.d/20-gettext.ini
    - /etc/php/7.4/cli/conf.d/20-iconv.ini
    - /etc/php/7.4/cli/conf.d/20-json.ini
    - /etc/php/7.4/cli/conf.d/20-phar.ini
    - /etc/php/7.4/cli/conf.d/20-posix.ini
    - /etc/php/7.4/cli/conf.d/20-readline.ini
    - /etc/php/7.4/cli/conf.d/20-shmop.ini
    - /etc/php/7.4/cli/conf.d/20-sockets.ini
    - /etc/php/7.4/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.4/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.4/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.4/cli/conf.d/20-tokenizer.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-dom` to temporarily ignore these required extensions.
ubuntu@ubuntu:~/webonary$ composer update
Composer is operating significantly slower than normal because you do not have the PHP curl extension enabled.
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - phpoffice/phpspreadsheet[1.17.1, ..., 1.24.1] require ext-dom * -> it is missing from your system. Install or enable PHP's dom extension.
    - Root composer.json requires phpoffice/phpspreadsheet ^1.17.1 -> satisfiable by phpoffice/phpspreadsheet[1.17.1, ..., 1.24.1].

To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.4/cli/php.ini
    - /etc/php/7.4/cli/conf.d/10-opcache.ini
    - /etc/php/7.4/cli/conf.d/10-pdo.ini
    - /etc/php/7.4/cli/conf.d/20-calendar.ini
    - /etc/php/7.4/cli/conf.d/20-ctype.ini
    - /etc/php/7.4/cli/conf.d/20-exif.ini
    - /etc/php/7.4/cli/conf.d/20-ffi.ini
    - /etc/php/7.4/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.4/cli/conf.d/20-ftp.ini
    - /etc/php/7.4/cli/conf.d/20-gettext.ini
    - /etc/php/7.4/cli/conf.d/20-iconv.ini
    - /etc/php/7.4/cli/conf.d/20-json.ini
    - /etc/php/7.4/cli/conf.d/20-phar.ini
    - /etc/php/7.4/cli/conf.d/20-posix.ini
    - /etc/php/7.4/cli/conf.d/20-readline.ini
    - /etc/php/7.4/cli/conf.d/20-shmop.ini
    - /etc/php/7.4/cli/conf.d/20-sockets.ini
    - /etc/php/7.4/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.4/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.4/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.4/cli/conf.d/20-tokenizer.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-dom` to temporarily ignore these required extensions.
```

Reply:
Added doc on PHP and composer installation at the top of the file.

Reply:
I'm not seeing the updates. Can you upload them again? Thanks.

Reply:
@johntseng I just set up a new laptop. This was my install for PHP 7.4, I think I have everything here. Webonary runs. You probably won't need lynx or redis.
```
sudo apt-get install php7.4 php7.4-mysqlnd php7.4-bcmath php7.4-gd php7.4-mbstring php7.4-zip php7.4-bcmath php7.4-json php7.4-curl php7.4-dom php7.4-simplexml php7.4-soap php7.4-dev libapache2-mod-php7.4 php7.4-cgi git libmcrypt-dev gcc re2c lynx libyaml-dev php-pear php7.4-xdebug php7.4-intl php7.4-redis
sudo pecl channel-update pecl.php.net
sudo pecl install yaml
sudo pecl install mcrypt
```

Reply:
@johntseng Sorry, I forgot to commit my latest changes. Please see below, and re-review. Thanks!